stdlib/public/core/StringUTF16View.swift

```diff
@@ -201,6 +204,14 @@ extension String.UTF16View: BidirectionalCollection {
       return _foreignIndex(i, offsetBy: n)
     }
 
+    let threshold = (
+      i == startIndex ? _breadcrumbStride / 2 : _breadcrumbStride)
+    if n.magnitude < threshold, !_guts.isASCII {
```

Comment:
is the sense of this condition right? Isn't direct computation always faster when `_guts.isASCII` is true? What am I misreading?

Reply:
The `_nativeGetIndex`/`_nativeGetOffset` calls below have O(1) ASCII fast paths, so calling the advancing loop in `_index(_:offsetBy:)` would make that case worse. (I used to have a special case for `_guts.isASCII` inside this branch, but it seems simpler to just let the original code take care of it.)